packages/react-components/src/components/chart/hooks/useChartsLegend.tsx

```diff
@@ -47,10 +54,26 @@ const LegendCell = (e: { datastream: DataStream; lineColor: string; name: string
     </div>
   ));
 
+  useEffect(() => {
```

Comment:
Is it necessary to use `useEffect` here? Can the name truncation be calculated each render, instead of using state?

Reply:
updated. removed the useEffect and managed using variable.